portal/server/helpers/constructPayload.js

```diff
@@ -5,16 +5,24 @@
  * @param {Object} body Request body (req.body)
  * @param {Array} properties Interested properties to be added into payload
  * @param {Boolean} csrf Include CSRF token, defaulted to `true`
+ * @param {Boolean} canPropertyBeEmpty Whether property can be empty in a payload
  * @returns {Object} Payload
  */
-const constructPayload = (body, properties, csrf = true) => {
+const constructPayload = (body, properties, canPropertyBeEmpty = false, csrf = true) => {
   let payload = {};
-
+  const bodyCopy = { ...body };
   // Return empty payload upon void mandatory arguments
   if (!body || !properties) {
     return payload;
   }
 
+  // Remove empty properties if flag is set to true, for example currency field, and MongoDB doesn't allow type change.
+  for (const property of properties) {
+    if (canPropertyBeEmpty === false && bodyCopy[property] === '') {
```

Comment:
if (!canPropertyBeEmpty && !bodyCopy[property])

Reply:
- on line 12 `payload` name is already used. Please suggest another name for `bodyCopy`
- When used `if (!canPropertyBeEmpty && !bodyCopy[property])` unit tests fail so I changed to i`f (!canPropertyBeEmpty && bodyCopy[property] === '')` and unit test are passing.